### PR TITLE
feat: Implement minimum TUI size 

### PR DESCRIPTION
**Pull request by jeevithakannan2**
# Pull Request

## Title
Implement minimum TUI size

## Type of Change
- [x] New feature
- [x] UI/UX improvement

## Description
- Display a warning message if the terminal size is small
- Prevents entries from not appearing on screen. Refer issue https://github.com/ChrisTitusTech/linutil/issues/242
- The method to resolve the issue is to render the TUI only when the size of TUI is above a certain threshold

warning

## Impact
- Disables users from accessing the UI when the terminal size is small 

## Issue related to PR
- Resolves https://github.com/ChrisTitusTech/linutil/issues/242

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Review by adamperkowski — APPROVED** [review]

File: tui/src/state.rs [hunk 1]
```diff
@@ -187,6 +232,19 @@ impl AppState {
     }
 
     pub fn handle_key(&mut self, key: &KeyEvent) -> bool {
+        // This should be defined first to allow closing
+        // the application even when not drawable ( If terminal is small )
+        if matches!(self.focus, Focus::TabList | Focus::List) {
+            if key.code == KeyCode::Char('q') {
```
Comment (adamperkowski) [inline tui/src/state.rs:238]:
```suggestion
            if key.code == KeyCode::Char('q')
                || key.code == KeyCode::Char('c') && key.modifiers.contains(KeyModifiers::CONTROL)
            {
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]

**Review by adamperkowski — APPROVED** [review]

**Review by nnyyxxxx — APPROVED** [review]

**Comment by nnyyxxxx** [comment]
Nice! bashtop also has a feature like this.
image

**Comment by jeevithakannan2** [comment]
I see, I think i should change Terminal size too small to current size. What do you think ?

**Comment by nnyyxxxx** [comment]
> I see, I think i should change Terminal size too small to current size. What do you think ?

No the current version is fine how it is.

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**